Floofstation V25

Memeji opened the pull request:
# Description

Floofstation V25.
Includes:

- #705 
- #900 (Reverted)
- #972 
- #985 
- #995 
- #997 
- #999 
- #1000 
- #1001 
- #1002 
- #1003 
- #1006 
- #1008 
- #1009 
- #1010 
- #1012 
- #1016 
- #1017 
- #1020 
- #1023 
- #1026 
- #1030 
- #1033 
- #1034 
- #1036 
- #1038 
- #1041 
- #1042

Mnemotechnician committed:
Fix some slot blocking issues (#1000)

Commit:
Automatic Changelog Update (#1000)

Commit:
Maybe Fix Antag Related Crash (#1523) (#1002)

# Description

I'm trying to fix a crash caused by character profiles saving the same
antag preference twice. I have no clue why or how it's happening, but
this should probably prevent it from happening by making the profile
sanitizer remove duplicate keys before saving them to the database.

The relevant crash log: 

[goobcrash.txt](https://github.com/user-attachments/files/18391763/goobcrash.txt)

# Changelog

:cl:
- fix: Attempted to fix a crash related to antag preferences saving
multiple times.

Co-authored-by: VMSolidus <evilexecutive@gmail.com>
Co-authored-by: sleepyyapril <123355664+sleepyyapril@users.noreply.github.com>

Commit:
Automatic Changelog Update (#1002)

fenndragon committed:
Sprite Movement Working with AIs (#972)

* Sprite Movement working with AI movement (#33494)

* FINALLY

* Update animals.yml

* sprit

* Update animals.yml

* Update animals.yml

* fix

* Update animals.yml

* Update animals.yml

---------

Co-authored-by: Ed <96445749+TheShuEd@users.noreply.github.com>

Commit:
Automatic Changelog Update (#972)

M3739 committed:
istg (#997)

Raikyr0 committed:
Change to Hypno Trait (#1009)

* Update mental.yml

* Update psionics.yml

* Update hypno.ftl

Commit:
Automatic Changelog Update (#1009)

Commit:
AddTraitSpecial (#1528) (#1003)

Adds a way to specify traits in the jobs special area.

Hashset so you can specify multiple traits. Tested on Nuclear14 to give
the tribals a specific language in a friendly way.
```
  special:
  - !type:AddTraitSpecial
    traits: [ LanguageTribal, LanguageChinese ]
```

Co-authored-by: Peptide90 <78795277+Peptide90@users.noreply.github.com>

Aikakakah committed:
Upstream Marking Updates (#999)

* Comically Large Markings Port From Impstation + Couple Fixes (#1852)

Comically large PR where I add a bunch of Impstation marking along with
a bunch more of fixes.

<!--
This is a semi-strict format, you can add/remove sections as needed but
the order/format should be kept the same
Remove these comments before submitting
-->

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Basically what this PR does is, with permission of the people from the
great server of Impstation, it adds most of the markings they've got
there, along with a couple personal fixes that I felt were needed, which
I will explain in the changelog.

---

<!--
A list of everything you have to do before this PR is "complete"
You probably won't have to complete everything before merging but it's
good to leave future references
-->

- [ ] Task
- [x] Completed Task

---

<!--
This is default collapsed, readers click to expand it and see all your
media
The PR media section can get very large at times, so this is a good way
to keep it clean
The title is written using HTML tags
The title must be within the <summary> tags or you won't see it
-->

<details><summary><h1>Media</h1></summary>
<p>

![Example Media Embed](https://example.com/thisimageisntreal.png)

</p>
</details>

---

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Many markings from Impstation.
- tweak: Marking limits have been expanded, now most races should be
able to have 6 markings instead of 1 or 2 or 3.
- fix: Most races should now have access to underwear.
- remove: My sanity.

<!-- This is an auto-generated comment: release notes by coderabbit.ai
-->

- **New Features**
- Expanded character customization with a broad range of new appearance
options including diverse markings, tattoos, scars, and hairstyles
across multiple species.
- Extended wardrobe support, now offering undershirt and underwear
options for additional species.
- Enhanced balance in visual customization by increasing point
allocations for many body parts.
- Introduced animated texture support to deliver more dynamic, fluid
visual effects for character designs.
- **New Markings**
- Added numerous new markings for various species, enhancing
customization options, including hair styles and various body markings.
- **New Localization Entries**
- Introduced new localization strings for hair accessories, markings,
and tattoos to support diverse character appearances.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

---------

Signed-off-by: BloodfiendishOperator <141253729+Diggy0@users.noreply.github.com>

* Alternative Bunny Ears (#1577)

Adds an alternative set of Bunny Ears and Bunny Tail markings for humans
as part of the Kemonomimi markings set, which comes from the bunny ears
item.

![SirinaBunny](https://github.com/user-attachments/assets/2c024ab0-ca8e-46e2-ab13-7bb71042786e)

![BunnyGirlEars](https://github.com/user-attachments/assets/9402700a-a7bd-4874-b505-8ddbdc6f51c9)

![image](https://github.com/user-attachments/assets/d73f0528-c625-4848-b1da-7f7085accd07)

![image](https://github.com/user-attachments/assets/a93f5424-5155-48d7-933c-c13693bcbc5b)

:cl: Skubman
add: Humans have received alternative bunny ears and bunny tail markings
that look the same as the bunny ears from the AutoDrobe, but
color-customizable.

* Markings From Impstation 2 (#1925)

what the funny title says

<!--
This is a semi-strict format, you can add/remove sections as needed but
the order/format should be kept the same
Remove these comments before submitting
-->

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

Adds some markings from Impstation (Again, who would have thought it'd
be Impstation again?), ports the merciful IPC wings from Goob LRP, adds
some markings to arachnids, a bunch of markings for Reptilians, and a
new hairstyle.

---

<!--
A list of everything you have to do before this PR is "complete"
You probably won't have to complete everything before merging but it's
good to leave future references
-->

- [ ] Task
- [x] Completed Task

---

<!--
This is default collapsed, readers click to expand it and see all your
media
The PR media section can get very large at times, so this is a good way
to keep it clean
The title is written using HTML tags
The title must be within the <summary> tags or you won't see it
-->

<details><summary><h1>Media</h1></summary>
<p>

</p>
</details>

---

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl: Diggy
- add: New markings for arachnids, reptilians.
- tweak: Allows IPCs to pick up overlay markings such as the gauze
stuff, Onis can now have the curved horn marking humans can have.

* Update markingsbundle2.ftl

* Update misc2.yml

---------

Signed-off-by: BloodfiendishOperator <141253729+Diggy0@users.noreply.github.com>
Co-authored-by: Skubman <ba.fallaria@gmail.com>

Cprn-Sprkl committed:
Coo Coo for Harpies (#1010)

* Add Coo Emote for Harpies

Adds five CC0 dove cooing sound effects so harpies can coo.

* Add missing comment

Added a comment to mark this as a floof addition,

* Fix license format

Little tweak to the license to pass the test.

TheRealMasterChief117 committed:
inital (#1008)

Commit:
Automatic Changelog Update (#1008)

VividPups committed:
Implant Trackers Rework (#705)

* Update cargo_armory.yml

* Dont merge yet still problems

Dont merge yet there is still Problems with crafting

* Researchable instead

* requested changes

Changes

Commit:
Automatic Changelog Update (#705)

M3739 committed:
Kitsune Hotfix (Give Topicals BiologicalMetaphysical Damage Containers) (#1006)

* Give topicals the `BiologicalMetaphysical` container

* Update comments to reflect PR number

* Update cybereyes to also have `BiologicalMetaphysical`

Commit:
Automatic Changelog Update (#1006)

ShadesMars committed:
Shadekin Rework (#995)

* Startup / .yml changes

* progress

* progress

* shadekin progress!

* LightExposure fully finished

* Apply suggestions from code review

* Apply Fixes

* Ship it

* Patch Phase

* Ship it!

* Bunch of fixes and addtions

* DarkLight

* doing uquick test

* Portal Fix + Improvement

* Finishing Touch

* Tiny haze fix

* yml happy :)

* Missing Category

* Build Fixes

* PortalLoadFix

* Dark Respawn Timer

* Better Predicted Collision Checks

* linter fixes

* Less Spam Mood / Tiny Improvements

* Oops, all psionic

* DarkTiles Improvement + FOLDERS§ WOW

Commit:
Automatic Changelog Update (#995)

Raikyr0 committed:
Update physical.yml (#1012)

Commit:
Automatic Changelog Update (#1012)

WhiteLightningTwister committed:
Adjust Job Counts & Adjusted Departments (#1016)

* Adjust Job Counts

Adds back the following things.
Justice job slots. (I.E, Chief justice, clerk, prosecutors & attorneys. )
Silion job slots. (I.E, cyborgs, medical cyborgs. )
Epistemics job slots. (I.E, robotics.)
Security job slots. (I.E, prison guards.)
Medical job slots. (I.E, psychologist. )

Increased max players from 35 > 60.

* Updated Parts of saltern.

Added job spawns.
Likewise some minor additions & fixes.

Commit:
Automatic Changelog Update (#1016)

Commit:
Upstream Merge #1521 | Nerf Salvage (#1001)

* Nerf Salvage (#1521)

I may have been a little *too* generous with Salvage players. This
update fixes the salvage guns spawning with a magazine or lethal ammo.
There was also an issue that they were perhaps "Too easy to obtain". In
actual practice, it turns out that a single "Mining Expedition" earns
salvage enough mining points to buy like 4 rifles. Which led to
situations where Salvage functionally became station security after they
then broke into the armory to print ammo.

Another issue I discovered is that Salvage can also use their breaching
charges to blow open security crates(And also blow open the
armory......). So now they don't spawn with bombs in their lockers, and
need to pay for them with said mining points.

<details><summary><h1>Media</h1></summary>
<p>

![image](https://github.com/user-attachments/assets/a5d3fba3-33b3-4f65-ab69-ee6d0f617633)

</p>
</details>

:cl:
- add: Added a legality notice to the description of salvage's firearms.
- tweak: The mining points cost of all salvage firearms have been
increased by 4x
- remove: Salvage no longer starts with bombs in their lockers. Buy them
with your mining points.

* Fix SVT Desc

Make the SVT Description match the others as per VividPups's request.

* Removed Seclite from salvage locker

* Update salvage_points.yml

removed gestio as an option.

---------

Co-authored-by: VMSolidus <evilexecutive@gmail.com>
Co-authored-by: Memeji Dankiri <39416235+Memeji@users.noreply.github.com>

Commit:
Automatic Changelog Update (#1001)

Mnemotechnician committed:
Update shuttle.yml (#1020)

Commit:
Automatic Changelog Update (#1020)

Raikyr0 committed:
Upstream Merge #2399 | Garand Ping (#1023)

* Add files via upload

* Update armory.yml

* Update lathe.yml

* Add files via upload

* Update snipers.yml

* Update security.yml

* Update tags.yml

* Update lathe.yml

* Update snipers.yml

Commit:
Automatic Changelog Update (#1023)

Memeji committed:
Merge branch 'stable' into master

Raikyr0 committed:
Update PowerCellSystem.Draw.cs (#900)

Raikyr0 committed:
Sex Update - Loadout (#1017)

* Update itemsLewd.yml

* Update items.yml

* Update itemsLewd.yml

* Update itemsLewd.yml

* Update funGroup.yml

Commit:
Automatic Changelog Update (#1017)

Raikyr0 committed:
Update specific.yml (#1033)

Commit:
Automatic Changelog Update (#1033)

ItsDzordz committed:
Clown Abominations Part 1 [Clown Rat] (#985)

* Clown rat is born

All the malice of the trickery of the honk mother made manifest

* Clown Rat for the Syndicate

Added a syndicate version of the clown rat, to the uplink.
Added the regular clown rat to the vent spawn event.

* Bug fix

Fixed the critter event error

* Bug fix #2

Correcting errors from bot checks

* Bug fix 3

Removing non existent damge type

* Value tweak

* Update Resources/Locale/en-US/_Floof/store/uplink-catalog.ftl

Co-authored-by: M3739 <47579354+M3739@users.noreply.github.com>

* Update Resources/Prototypes/_Floof/Entities/Mobs/NPCs/clown_abominations.yml

Co-authored-by: M3739 <47579354+M3739@users.noreply.github.com>

* burn inhertance bug fix

* Update events.yml

* Price fix

* Price fix 2

* ash amount fix

* price fix 3

* Price fix 5

* Price fix 6

* Test re-run

teruning test since it seems the current error is irrelevent to this pr

* Faction fix

added the overwritten faction from the parnets so they are attacked by shiva and laika

---------

Co-authored-by: M3739 <47579354+M3739@users.noreply.github.com>

Commit:
Automatic Changelog Update (#985)

M3739 committed:
Task Kitsunebi (Fix Foxfires, Locale) (#1030)

* "Fix" charges and limit fox fires

* Missing Comment

* Fix locale

* Make it better ig

* Update comments and PR number

Commit:
Automatic Changelog Update (#1030)

Mnemotechnician commented:
There is currently a bug caused by #900 that causes batteries (specifically those inside PowerCellDraw-ing entities) to be drained 60 or 30 times slower than normally. Just as I said before. This is the battery of a translator after 2 minutes of usage:
![image](https://github.com/user-attachments/assets/26de8f81-41cd-4f60-88db-dc7077658246)

Mnemotechnician committed:
Revert "Update PowerCellSystem.Draw.cs (#900)" (#1036)

This reverts commit 4a38106b8f25feb09589de801bb26625369cfce0.

Commit:
Automatic Changelog Update (#1036)

Memeji committed:
Toy Sword Fix (#1026)

* Fix JobCourier to JobMailCarrier.

* Fixes the Toy Sword sprite

Commit:
Automatic Changelog Update (#1026)

Mnemotechnician reviewed (approved):
Just realized I forgot to approve this. Oops.

Mnemotechnician committed:
Fix CPR Sound Looping (#1034)

* Disable CPR sound looping

* Actual fix

Commit:
Automatic Changelog Update (#1034)

Lawdog4817 committed:
Accentless (#1038)

Commit:
Automatic Changelog Update (#1038)

Mnemotechnician committed:
Fix harpy coos (#1041)

Commit:
Automatic Changelog Update (#1041)

Mnemotechnician committed:
Fix duplicate lathe recipes (#1042)

Commit:
Automatic Changelog Update (#1042)

Raikyr0 committed:
Energy Toys (#1035)

* Update toys.yml

* Update esheath.yml

* Update uplink_catalog.yml

* Update uplink-catalog.ftl

* Update esheath.yml

Commit:
Automatic Changelog Update (#1035)

TheRealMasterChief117 committed:
Alternate Female Lizard Chest Fixes, Sprite Change, and Reorganization (#1025)

* inital

* should be good now

* yuh

Commit:
Automatic Changelog Update (#1025)

Aikakakah committed:
Clothing Updates (#998)

* Port Adjustable Jumpsuits And Undergarments (#1720)

Ports https://github.com/impstation/imp-station-14/pull/375
and https://github.com/impstation/imp-station-14/pull/384
Which are meant to go together.

<details><summary><h1>Media</h1></summary>
<p>

![image](https://github.com/user-attachments/assets/090482f0-3665-42c6-badd-0985a928adfd)

</p>
</details>

:cl:
- add: Added the ability to roll down jumpsuits to the waist, as well as
undergarment markings for wearing alongside said rolled jumpsuits.

---------

Co-Authored-By: V <97265903+formlessnameless@users.noreply.github.com>

* Upstream #1720

Co-Authored-By: V <97265903+formlessnameless@users.noreply.github.com>

* The Magmantle! (#2295)

<!--
This is a semi-strict format, you can add/remove sections as needed but
the order/format should be kept the same
Remove these comments before submitting
-->

<!--
Explain this PR in as much detail as applicable

Some example prompts to consider:
How might this affect the game? The codebase?
What might be some alternatives to this?
How/Who does this benefit/hurt [the game/codebase]?
-->

This PR adds the Magmantle, a mantle that functions like magboots, but
slightly worse (like 15% more slowdown, but it does provide 20%
temperature protection like winter boots or scarves). Its intended for
species without the feet slot, which are harpy, diona and arachne. It is
available in the species tab of loadouts, for 6 points. Only Security,
Command, Dignitaries, Engineering, Salvies and Parameds can take it, if
they are a harpy, diona or arachne.

---

<!--
This is default collapsed, readers click to expand it and see all your
media
The PR media section can get very large at times, so this is a good way
to keep it clean
The title is written using HTML tags
The title must be within the <summary> tags or you won't see it
-->

<details><summary><h1>Media</h1></summary>
<p>

![mmantle](https://github.com/user-attachments/assets/782bef02-cba5-4d69-86f0-eece4aa3ea1a)

![mmantledesc](https://github.com/user-attachments/assets/455b3ede-8467-4aa9-a668-b266a2abde38)
![Magmantle working, speed modifiers
showcase](https://github.com/user-attachments/assets/bf03525d-3d8a-410b-bd3c-e1a9cdb0d1d5)

</p>
</details>

---

<!--
You can add an author after the `:cl:` to change the name that appears
in the changelog (ex: `:cl: Death`)
Leaving it blank will default to your GitHub display name
This includes all available types for the changelog
-->

:cl:
- add: Added the Magmantle, a mantle that functions as worse magboots on
the neck slot (15% more slow, but they give 20% temperature protection
like winter boots or scarves). It is available in the species section of
loadouts, for all the species that do not have a feet slot. (Harpy,
Diona, Arachne)

---------

Co-authored-by: VMSolidus <evilexecutive@gmail.com>

* Hi-Viz Now Glows in the Dark (#2183)

Port Hi-viz now glows in the dark from
[wizards](https://github.com/space-wizards/space-station-14)
Pr: https://github.com/space-wizards/space-station-14/pull/34087

---

<details><summary><h1>Media</h1></summary>
<p>

![Captura de tela 2025-04-06
112220](https://github.com/user-attachments/assets/38fbe984-f36c-4b9c-8353-1619471861e4)

</p>
</details>

---

:cl: kosticia and Will-Oliver-Br
- tweak: Hi-viz now glows in the dark

Co-authored-by: kosticia <kosticia46@gmail.com>

* Revert "The Magmantle! (#2295)"

This reverts commit b97ba11c597f3a445418ef6d924603331dd88f10.

---------

Co-authored-by: V <97265903+formlessnameless@users.noreply.github.com>
Co-authored-by: VMSolidus <evilexecutive@gmail.com>
Co-authored-by: Will-Oliver-Br <164823659+Will-Oliver-Br@users.noreply.github.com>
Co-authored-by: kosticia <kosticia46@gmail.com>

Commit:
Upstream Merge #1534 | Fix NTR Stamp Icon On Paper & Fix Job IDs (#1004)

* Fix NTR Stamp Icon On Paper & Fix Job IDs (#1534)

# Description
This fixes the Job IDs that show in the time transfer panel because the
users that added them did not add the FTL for them to be named in game.

---

# Media
Images are broken up by their respective showings.

<details><summary><h3>Job Icon Images</h3></summary>
<p>


![image](https://github.com/user-attachments/assets/f7356d10-5a82-4cc1-9589-3a6099920c5b)

![image](https://github.com/user-attachments/assets/0a7423e0-a1b9-429a-85b5-0049c6891e0e)

![image](https://github.com/user-attachments/assets/8333ad4f-475a-4344-81b7-184fbffc47d8)

</p>
</details>
<details><summary><h3>Nanotrasen Representative Stamp</h3></summary>
<p>


![image](https://github.com/user-attachments/assets/9a7b9693-d9bd-4d06-b6df-fa0aa3787096)

</p>
</details>

---

# Changelog
:cl:
- fix: Fixed the NTR Stamp Texture on Paper
- fix: Fixed name of Senior Officer, CBURN, and Deathsquad.

* Remove duplicate ftl

Already have Senior Officer on line 7.

Co-authored-by: FoxxoTrystan <45297731+FoxxoTrystan@users.noreply.github.com>

---------

Co-authored-by: John Willis <143434770+CerberusWolfie@users.noreply.github.com>
Co-authored-by: Memeji Dankiri <39416235+Memeji@users.noreply.github.com>
Co-authored-by: FoxxoTrystan <45297731+FoxxoTrystan@users.noreply.github.com>

Mnemotechnician committed:
Id Lockable Airlocks (#1029)

* Id lockable airlocks

* Fix inheritance

Commit:
Automatic Changelog Update (#1004)
